constraint/pkg/client/client.go

```diff
@@ -135,29 +166,72 @@ func (c *Client) AddTemplate(ctx context.Context, templ *templates.ConstraintTem
 		return resp, err
 	}
 
-	if err := c.driver.AddTemplate(ctx, templ); err != nil {
+	newDriverN := c.driverForTemplate(templ)
+
+	driver, ok := c.drivers[newDriverN]
+	if !ok {
+		return resp, fmt.Errorf("%w: available drivers: %v, wanted %q", clienterrors.ErrNoDriver, c.driverPriority, c.driverForTemplate(templ))
+	}
```

Comment:
Then we wouldn't be able to use the return value of driverForTemplate() to generate an error string below.

We also would not be able to run a comparison to see if the driver changed (strings can always be trivially compared, structs may not be).

Reply:
You could call driver.Name(), but I really don't see any benefit to doing it the way you suggest.

Reply:
>  to generate an error string below.

we could wrap the err returned by `driverForTemplate` to add more contextual information as needed? 

I think `driverForTemplate` is misleading since we actually return the language/ name of the driver out of that func and not the `driver` struct/ object itself. Maybe `diverNameForTemplate` otherwise?

whatever you choose here is fine w me

Reply:
I think it is not worth the time to bikeshed on the naming of purely internal functions.